src/capabilities.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {UAParser} from 'ua-parser-js';
+
+/**
+ * A browser feature.
+ */
+export type Capability =
+    // ECMAScript 2015 (aka ES6).
+    'es2015' |
+    // HTTP/2 Server Push.
+    'push';
+
+/**
+ * Return a capability map for the given user agent string.
+ */
+export function capabilities(userAgent: string):
+    {[key in Capability]: boolean} {
+  const ua = new UAParser(userAgent);
+  const supports = browserCapabilities[ua.getBrowser().name];
```

Comment:
How about a function like:

```ts
function testPredicates<K, T>(predicates: Map<K, (T) => boolean>, o: T): Set<K> {
  const passedTests = [];
  for (const [key, predicate] of predicates.entries()) {
    if (predicate(o)) {
      passedTests.push(key);
    }
  }
  return new Set(passedTests);
}
```
and 

```ts
return testPredicates(browserCapabilities.get(ua.getBrowser().name) || new Map(), ua);
```

This doesn't make it any shorter obviously, but it supports any predicates in the configuration.